numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -557,6 +557,37 @@ NPY_INPLACE @type@ npy_cbrt@c@(@type@ x)
     }
 }
 
+NPY_INPLACE @type@
+npy_sinc@c@(@type@ x)
+{
+    @type@ r;
+
+    if (npy_isnan(x)) {
+	return (@type@) NPY_NAN;
+    }
+    else if (npy_isinf(x)) {
+	return 0;
+    }
+    else if (x == 0) {
+	return 1;
+    }
+
+    if (x < 0) {
+	x = -x;
+    }
+
+    r = npy_fmod@c@(x, 2);
```

Comment:
If you have any reference as to this method, I would be very happy about a comment in the function pointing to it.

Reply:
Hm various related implementations for `sinpi` exist:

- Go: https://github.com/golang/go/blob/50bd1c4d4eb4fac8ddeb5f063c099daccfb71b26/src/math/lgamma.go#L324

- Mpmath: https://github.com/fredrik-johansson/mpmath/blob/2858b1000ffdd8596defb50381dcb83de2bcccc6/mpmath/math2.py#L113

- Boost: https://github.com/boostorg/math/blob/c759981a2af05c7319c07e532e070e59a72cc74f/include/boost/math/special_functions/sin_pi.hpp#L23

- SciPy (which doesn't really count because I wrote it): https://github.com/scipy/scipy/blob/master/scipy/special/_trig.pxd#L16

Preferences for any of those? Unfortunately I don't know of a paper with a reference for `sinpi`-the technique is likely too minor to merit publication.